torchmetrics/retrieval/recall_precision.py

```diff
@@ -0,0 +1,185 @@
+# Copyright The PyTorch Lightning team.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+from functools import partial
+from typing import Any, Dict, Optional, Tuple
+
+import torch
+from torch import Tensor, tensor
+
+from torchmetrics import Metric
+from torchmetrics.retrieval import RetrievalPrecision, RetrievalRecall
+from torchmetrics.utilities.checks import _check_retrieval_inputs
+from torchmetrics.utilities.data import get_group_indexes
+
+
+class MinPrecisionError(Exception):
```

Comment:
lest raise it as ValueError

Reply:
Fixed by func `_retrieval_recall_at_fixed_precision`, which catches all ValueErrors :)